python/paddle/onnx/export.py

```diff
@@ -0,0 +1,107 @@
+
+
+def export(layer, path, input_spec=None, opset_version=9, **configs):
+    """
+    Export Layer as ONNX format model, which can be used for inference.
+    Now, it supports a limited operater set and dynamic models.(e.g., MobileNet.)
+    More features and introduction, Please reference `Github <https://github.com/PaddlePaddle/paddle2onnx>`_ .
+
+    Args:
+        layer (Layer): The Layer to be exported.
+        path (str): The path prefix to export model. The format is ``dirname/file_prefix`` or ``file_prefix`` ,
+            and the exported ONNX file suffix is ``.onnx`` . 
+        input_spec (list[InputSpec|Tensor], optional): Describes the input of the exported model's forward 
+            method, which can be described by InputSpec or example Tensor. If None, all input variables of 
+            the original Layer's forward method would be the inputs of the exported ``ONNX`` model. Default None.
+        opset_version(int, optional): Opset version of exported ONNX model.
+            Now, stable supported opset version include 9, 10, 11. Default 9.
+        **configs (dict, optional): Other export configuration options for compatibility. We do not 
+            recommend using these configurations, they may be removed in the future. If not necessary, 
+            DO NOT use them. Default None.
+            The following options are currently supported:
+            (1) output_spec (list[Tensor]): Selects the output targets of the exported model.
+            By default, all return variables of original Layer's forward method are kept as the 
+            output of the exported model. If the provided ``output_spec`` list is not all output variables, 
+            the exported model will be pruned according to the given ``output_spec`` list. 
+    Returns:
+        None
+    Examples:
+        .. code-block:: python
+
+            import paddle
+            import numpy as np
+
+            class LinearNet(paddle.nn.Layer):
+                def __init__(self):
+                    super(LinearNet, self).__init__()
+                    self._linear = paddle.nn.Linear(128, 10)
+
+                def forward(self, x):
+                    return self._linear(x)
+
+            # export model with InputSpec, which supports set dynamic shape for inputs.
+            def export_linear_net():
+                model = LinearNet()
+                x_spec = paddle.static.InputSpec(shape=[None, 128], dtype='float32')
+                paddle.onnx.export(model, 'linear_net', input_spec=[x_spec])
+
+            export_linear_net()
+
+
+            class Logic(paddle.nn.Layer):
+                def __init__(self):
+                    super(Logic, self).__init__()
+
+                def forward(self, x, y, z):
+                    if z:
+                        return x
+                    else:
+                        return y
+
+            # export model with Tensor, which supports prune model by set 'output_spec' with output of model.
```

Comment:
Export model with `paddle.Tensor` as `input_spec`, and model output as `output_spec`, onnx exporter will automatically prune out the inference model.